esp/useful_scripts/csv_fresh_classroom_import.py

```diff
@@ -68,12 +84,94 @@ def get_available_furnishings():
     for res, idx in RESOURCE_MATCHING.iteritems():
         furnish_name = furnish_headers[idx] if idx is not None else "None"
         print "{}: {} (Column {})".format(res, furnish_name, str(idx))
-    conf = raw_input("Confirm correctness: y/n")
+    conf = raw_input("Confirm correctness: y/n ")
     if "y" in conf.lower():
         break
     else:
         RESOURCE_MATCHING = {}
 
+SPACE_TYPE_MATCHING = {}
+def match_space_type(space_type):
+    if space_type == "":
+        return ""
+    # http://stackoverflow.com/questions/3673434/find-subsequences-of-strings-within-strings
+    pattern = ".*".join(space_type)
+    matches = []
+    for possible_type in POSSIBLE_SPACE_TYPES:
+        if re.search(pattern, possible_type):
+            matches.append(possible_type)
+    if len(matches) == 1:
+        return matches[0]
+    else:
+        return None
+
+def get_possible_space_Types():
+    return [(i, space) for i, space in enumerate(POSSIBLE_SPACE_TYPES) if i not in
+            SPACE_TYPE_MATCHING.values()]
+
+force_manual = False
+SPACE_TYPE_IDX = None
+while True:
+    print "Now attempting to match space type header..."
+    space_header_candidates = ["space type", "classroom space"]
+    for i, header in enumerate(furnish_headers):
+        if header.lower() in space_header_candidates:
+            SPACE_TYPE_IDX = i
+            break
+    if SPACE_TYPE_IDX is None or force_manual:
+        print "Unable to automatically determine space type header."
+        print "Available columns:"
+        for f in get_available_furnishings():
+            print f
+        idx = raw_input("Input furnishing index (or 'None'): ")
+        if idx == "None":
+            SPACE_TYPE_IDX = None
+        else:
+            SPACE_TYPE_IDX = int(idx)
+    print "Header for the classroom space type data is {}".format(
+        furnish_headers[SPACE_TYPE_IDX])
+    conf = raw_input("Confirm correctness: y/n ")
+    if "y" in conf.lower():
+        break
+    else:
+        SPACE_TYPE_IDX = None
+        force_manual = True
+
+force_manual = False
+if SPACE_TYPE_IDX is not None:
+    space_types = set([row[SPACE_TYPE_IDX] for row in furnish_classrooms
+                      if row[FURNISH_ROOM_NUMBER_COL] in ALL_USED_ROOMS])
+    while True:
+        print "Now attempting to match space types..."
+        for space_type in space_types:
+            match_attempt = match_space_type(space_type)
+            if match_attempt is None or force_manual:
+                possible_types = get_possible_space_Types()
+                if force_manual:
+                    print "Manual mode forced due to attempt failure."
+                print ("Unable to automatically match space type "
+                "{} to possible space types.").format(space_type)
+                print "Available space types:"
+                for space in possible_types:
```

Comment:
May want to `for i, space in enumerate(possible_types): print i, space` to make typing the index easy.  (Similarly a couple other places.)

Reply:
It's already taken care of as part of get_possible_space_types() but I'll rewrite to make it clearer.